src/Storage/Token.php

```diff
@@ -0,0 +1,115 @@
+<?php
+/**
+ * This file is part of Lcobucci\JWT, a simple library to handle JWT and JWS
+ *
+ * @license http://opensource.org/licenses/BSD-3-Clause BSD-3-Clause
+ */
+
+declare(strict_types=1);
+
+namespace Lcobucci\JWT\Storage;
```

Comment:
I decided to extract an interface from the Token but definitely needs to be segregated (the only similarity between a nested token and a "normal" token is that they have headers).

`Storage` is a terrible name but I was out of option.
I thought about `Lcobucci\JWT\Token` but `Lcobucci\JWT\Token\Token` is weird.

Another option would be `Lcobucci\JWT\Token\Plain` (but it's also weird IMO) and `Lcobucci\JWT\Token\Encrypted`.

Any thoughts?